compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirDeclarationCheckerUtils.kt

```diff
@@ -168,8 +175,31 @@ internal fun checkPropertyAccessors(
     reporter: DiagnosticReporter,
     context: CheckerContext
 ) {
-    property.setter?.source?.let {
-        if (property.isVal) {
+    if (property.isVal) {
+        if (property.getter != null) {
+            var reassignment: FirBackingFieldReference? = null
+            val visitor = object : FirVisitor<Unit, FirStatement?>() {
+                override fun visitElement(element: FirElement, data: FirStatement?) {
+                    element.acceptChildren(this, data)
+                }
+
+                override fun visitVariableAssignment(variableAssignment: FirVariableAssignment, data: FirStatement?) {
+                    val backingFieldReference = variableAssignment.lValue as? FirBackingFieldReference
+                    if (backingFieldReference?.resolvedSymbol?.fir == property) {
+                        reassignment = backingFieldReference
```

Comment:
Is it intended that only the "last" reassignment gets reported, rather than reporting all of them?

Reply:
🤦  I didn't think of that.

And... interestingly, FE 1.0 reports the first violation, and here I matched that by recording the first backing field reference, too.